client/matchmaking/socket-handlers.js

```diff
@@ -162,8 +162,8 @@ const eventToAction = {
       mapData: { format },
       mapUrl,
     } = event.chosenMap
-    // NOTE(2Pac): We can't download map any sooner since we don't know which map we'll play until
-    // all players accept the game
+    // Even though we're downloading the whole map pool as soon as the player enters the queue,
+    // we're still leaving this as a check to make sure the map exists before starting a game.
     mapStore.downloadMap(hash, format, mapUrl)
```

Comment:
Half-unrelated to this PR, but shouldn't this have been made to await on the function to finish before continuing the launch?

Reply:
Yeah, I was thinking about that today myself.

We're not waiting for it in the lobbies and I think I just copy-pasted it from there when I initially added it here. But it's much more likely to cause issues here than downloading a map when someone joins a lobby ^^.

Not sure if this is the best place to await stuff for though. Maybe the active-game-manager should act as a fail-safe in case the map was not downloaded yet from these other places and wait for it to finish before launching the game?

Reply:
Yeah, this should ideally get passed into that workflow in some way so that the game doesn't get told to set up until the map download is finished. That can go into a separate PR though, but while you're here can you attach an error handler to this Promise and log failures?

Reply:
Should we do this in all the places where we call `downloadMap`˙?

Reply:
Ideally, yeah